Correct Rubocop for `private_class_method` method documentation.

bigzed opened the pull request:
The `private_class_method` modifier was not respected when checking whether a method needs documentation. I was able to fix that, but unable to fix the corresponding spec when documentation is present. Rubocop unfortunately still complains.

-----------------

Before submitting the PR make sure the following are checked:

* [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Ran `bundle exec rake default`. It executes all tests and runs RuboCop on its own code.
* [x] Added an entry (file) to the [changelog folder](https://github.com/rubocop/rubocop/blob/master/changelog/) named `{change_type}_{change_description}.md` if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format) for details.

[1]: https://chris.beams.io/posts/git-commit/

bbatsov commented:
> I was able to fix that, but unable to fix the corresponding spec when documentation is present. Rubocop unfortunately still complains.

Can you elaborate on that? What remains to be done exactly?

bigzed commented:
Sorry for the long delay. I removed the offending spec for now. The MR now only adds the `private_class_method` modifier to the non public modifiers and adds the corresponding spec.

The second commit I removed was testing whether a documented `private_class_method def blah` would register an offense and it unfortunately did. I'll revisit that and will submit a second as soon as I have fixed it.
Thank you and sorry for the delay.

bbatsov commented:
The changes look good. Don't forget to reflect them in the changelog.

bbatsov commented on CHANGELOG.md:
```diff
@@ -11,6 +11,11 @@
 
 ## master (unreleased)
 
+### Bug fixes
```
You shouldn't edit the changelog manually - it's generated from the files in the `changelog` folder at release time.

bigzed replied:
Sorry, my mistake. Is this the correct way?
